src/server/enterprise/server/api_server.go

```diff
@@ -332,3 +348,228 @@ func (a *apiServer) Deactivate(ctx context.Context, req *ec.DeactivateRequest) (
+	} else {
+		// Curiously, Get can return no error and an empty configmap!
+		// If this happens, need to set the name and namespace.
+		if c.ObjectMeta.Name == "" {
+			c = newPachdConfigMap(namespace)
+		}
+		if c.Data == nil {
+			c.Data = make(map[string]string)
+		}
+		if paused && c.Data["MODE"] == "paused" {
+			// Short-circuit and do not update if configmap
+			// is already in the correct state.  This keeps
+			// the updatedAt semantics correct when checking
+			// the pause status.
+			return nil
+		} else if !paused && c.Data["MODE"] != "paused" {
+			// Short-circuit and do not update if configmap
+			// is already in the correct state.  This keeps
+			// the updatedAt semantics correct when checking
+			// the pause status.
+			return nil
+		}
+		if paused {
+			c.Data["MODE"] = "paused"
+		} else {
+			c.Data["MODE"] = a.env.UnpausedMode
+		}
+		if c.Annotations == nil {
+			c.Annotations = make(map[string]string)
+		}
+		c.Annotations[updatedAtFieldName] = time.Now().Format(time.RFC3339)
+		if _, err := cc.Update(ctx, c, metav1.UpdateOptions{}); err != nil {
+			return errors.Errorf("could not update configmap: %w", err)
+		}
+	}
+
+	if err := retry.RetryOnConflict(retry.DefaultBackoff, func() error {
+		dd := kc.AppsV1().Deployments(namespace)
+		d, err := dd.Get(ctx, "pachd", metav1.GetOptions{})
+		if err != nil {
+			return err //nolint:wrapcheck
+		}
+		// Updating the spec rolls the deployment, killing each pod and causing
+		// a new one to start.
+		d.Spec.Template.Annotations["kubectl.kubernetes.io/restartedAt"] = time.Now().Format(time.RFC3339)
+		if _, err := dd.Update(ctx, d, metav1.UpdateOptions{}); err != nil {
+			return err //nolint:wrapcheck
+		}
+		return nil
+	}); err != nil {
+		return errors.Errorf("could not updated pachd deployment: %v", err)
+	}
+
+	return nil
+}
+
+func newPachdConfigMap(namespace string) *v1.ConfigMap {
+	return &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "pachd-config",
+			Namespace: namespace,
+		},
+		Data: map[string]string{"MODE": "full"},
+	}
+}
+
+func scaleDownWorkers(ctx context.Context, kc *kubernetes.Clientset, namespace string) error {
+	rc := kc.CoreV1().ReplicationControllers(namespace)
+	ww, err := rc.List(ctx, metav1.ListOptions{
+		LabelSelector: "suite=pachyderm,component=worker",
+	})
+	if err != nil {
+		return errors.Errorf("could not list workers: %v", err)
+	}
+	for _, w := range ww.Items {
+		if _, err := rc.UpdateScale(ctx, w.GetName(), &autoscalingv1.Scale{
+			Spec: autoscalingv1.ScaleSpec{
+				Replicas: 0,
+			},
+		}, metav1.UpdateOptions{
+			FieldManager: "enterprise-server",
+		}); err != nil {
+			return errors.Errorf("could not scale down %s: %v", w.GetName(), err)
+		}
+	}
+	return nil
+}
+
+func (a *apiServer) Unpause(ctx context.Context, req *ec.UnpauseRequest) (resp *ec.UnpauseResponse, retErr error) {
+	if a.env.mode == UnpausableMode {
+		return nil, errors.Errorf("this pachd instance is not in a pausable mode")
```

Comment:
change error message to "pachd is already unpaused."

Reply:
That is true but it is not an error to unpause a full-mode cluster; in this case the error really is that the cluster can neither be paused nor unpaused. Maybe it is clearer to state ‘this pachd instance is not in an unpausable mode’?

Reply:
works for me!